Shifty.App/Components/Voucher.razor

```diff
@@ -0,0 +1,138 @@
+@namespace Components
+@using System.ComponentModel.DataAnnotations
+@using Shifty.App.Services
+@using Shifty.Api.Generated.AnalogCoreV1
+@using Shifty.Api.Generated.AnalogCoreV2
+@inject IProductService _productService
+@inject IVoucherService _voucherService
+@inject ISnackbar Snackbar
+@inject IJSRuntime JSRuntime
+
+<MudContainer>
+    <MudCard Class="mb-auto" Width="40vw">
+        <MudCardContent>
+            <MudText Align="Align.Center" Class="mb-n4">Issue Voucher Form</MudText>
+            <MudForm @bind-IsValid="@_isFormValid" >
+                <MudSelect  T="ProductDto" 
+                            Label="Product"
+                            Placeholder="Available products..."
+                            AnchorOrigin="Origin.BottomCenter"
+                            ToStringFunc="@_converter"
+                            Required="true"
+                            RequiredError="Product is required"
+                            @bind-Value=_voucherForm.Product>
+
+                    @foreach (var product in _products)
+                    {
+                        <MudSelectItem Value="@product" />
+                    }
+                </MudSelect>
+
+                <MudNumericField @bind-Value="_voucherForm.Amount"
+                                Placeholder="1" 
+                                Label="Amount" 
+                                Variant="Variant.Text" 
+                                Required="true"
+                                RequiredError="Product is required" 
+                                Min="1"
+                                Max="10" />
+
+                <MudTextField T="string"
+                    @bind-Value="_voucherForm.Description"
+                    Label="Description" 
+                    Required="true"
+                    RequiredError="Description is required" />
+                
+                <MudCardActions>
+                    <MudButton  Variant="Variant.Filled"
+                                Color="Color.Primary"
+                                Class="ml-auto"
+                                Disabled="@(!_isFormValid)"
+                                OnClick="@(async () => await IssueVoucher())"
+                                EndIcon="@Icons.Material.Filled.Sell">
+                        Issue Voucher
+                    </MudButton>
+                </MudCardActions>
+            </MudForm>
+            @if (_showProgressBar)
+            {
+                <MudContainer class="d-flex justify-content-center">
```

Comment:
We a custom loading indicator we can consider using (LoadingIndicator.razor)

Reply:
Using our own loading indicator now :)